images/shared-images-controller/main.go

```diff
@@ -0,0 +1,156 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/containers/podman/v4/pkg/bindings"
+	"github.com/containers/podman/v4/pkg/bindings/images"
+	"github.com/google/go-github/github"
+	"github.com/sirupsen/logrus"
+	"kubevirt.io/project-infra/robots/pkg/querier"
+)
+
+var log *logrus.Logger
+
+func init() {
+	log = logrus.New()
+	log.SetOutput(os.Stdout)
+}
+
+func getLatestTag() (tag string, err error) {
+	resp, err := http.Get("https://raw.githubusercontent.com/kubevirt/kubevirt/main/cluster-up/version.txt")
+	if err != nil {
+		log.WithError(err).Errorf("Failed to get latest kubevirtci tag")
+		return "", err
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.WithError(err).Errorf("Reading latest kubevirtci tag failed")
+		return "", err
+	}
+	tag = string(body)
+	return strings.TrimSpace(tag), nil
+}
+
+func pullRequiredImages(ctx context.Context, tag string) error {
+	var versions []string
+
+	endpoint := "https://api.github.com/"
+	client, err := github.NewEnterpriseClient(endpoint, endpoint, nil)
+	if err != nil {
+		log.WithError(err)
+		return err
+	}
+	context := context.Background()
+	releases, _, err := client.Repositories.ListReleases(context, "kubernetes", "kubernetes", nil)
+	if err != nil {
+		log.WithError(err).Errorf("Failed to list releases from kubernetes/kubernetes")
+		return err
+	}
+	minors := querier.LastThreeMinor(uint(1), releases)
+	for _, minor := range minors {
+		r := querier.ParseRelease(minor)
+		releaseVer := fmt.Sprintf("%s.%s", r.Major, r.Minor)
+		versions = append(versions, releaseVer)
+	}
+
+	log.Infoln("Last three minor releases", versions)
+
+	imageNames := map[string]struct{}{}
+	kubevirtci_repo := "quay.io/kubevirtci/"
+
+	imageList, err := images.List(ctx, nil)
+	if err != nil {
+		log.WithError(err).Errorf("Failed to list images")
+		return err
+	}
+	for _, i := range imageList {
+		for _, imageName := range i.RepoTags {
+			imageNames[imageName] = struct{}{}
+		}
+	}
+
+	for _, version := range versions {
+		log.Infoln("Kubevirt Provider version: ", version)
+		name := fmt.Sprintf("%sk8s-%s:%s", kubevirtci_repo, version, tag)
+		_, exists := imageNames[name]
+		if exists {
```

Comment:
nit: you can do this in one line:

```suggestion
		if _, exists := imageNames[name]; exists {
```